Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java

```
@@ -118,13 +120,36 @@ protected void validateAssertion(AssertionWrapper assertion) throws WSSecurityEx
             List<ValidatorSuite> validators = new LinkedList<ValidatorSuite>();
             validators.add(org.opensaml.Configuration.getValidatorSuite("saml2-core-schema-validator"));
             validators.addAll(getSaml2SpecValidators());
-            
+
             for (AuthzDecisionStatement auth : assertion.getSaml2().getAuthzDecisionStatements()) {
                 if (StringUtils.isBlank(auth.getResource())) {
                     auth.setResource(TEMP_RESOURCE_FOR_VALIDATION);
                 }
             }
 
+            Issuer issuer = assertion.getSaml2().getIssuer();
+            String attributeNotRequired = null;
+            if (issuer.getFormat().equals("urn:oasis:names:tc:SAML:1.1:nameid-format:entity")) {
+                if (!StringUtils.isBlank(issuer.getSPProvidedID())) {
+                    attributeNotRequired = issuer.getSPProvidedID();
+                }
+                if (!StringUtils.isBlank(issuer.getNameQualifier())) {
+                    attributeNotRequired = issuer.getNameQualifier();
+                }
+
+                if (!StringUtils.isBlank(issuer.getSPNameQualifier())) {
+                    attributeNotRequired = issuer.getSPNameQualifier();
+
+                }
+            }
+            if (NullChecker.isNotNullish(attributeNotRequired)) {
+
+                throw new WSSecurityException("SOAP header element Security/Assertion/Issuer/@Format = " + ""
```

I think this error message is a little misleading. Can we just throw the exception 3 times from each of the above if conditions? That way you can specifically say SPProviderID (or NameQualifier, or SPNameQualifier) cannot be present if issuer format is entity.